parachain/pallets/omni-account/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn activate_identity(
+			origin: OriginFor<T>,
+			who: Identity,
+			identity: OmniAccountIdentity,
+		) -> DispatchResult {
+			let _ = T::TEECallOrigin::ensure_origin(origin)?;
+			let mut id_graph_links =
+				IDGraphs::<T>::get(&who).ok_or(Error::<T>::IdentityNotFound)?;
+			let id_graph_link = id_graph_links
+				.iter_mut()
+				.find(|(id, _)| id == &identity)
+				.ok_or(Error::<T>::IdentityNotFound)?;
+			id_graph_link.1.activate();
+
+			IDGraphs::<T>::insert(who.clone(), id_graph_links);
+
+			let identity_hash = H256::from(blake2_256(&identity.encode()));
+
+			Self::deposit_event(Event::IdentityActivated { who, identity: identity_hash });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn set_identity_networks(
+			origin: OriginFor<T>,
+			who: Identity,
+			identity: OmniAccountIdentity,
+			web3networks: Vec<Web3Network>,
+		) -> DispatchResult {
+			let _ = T::TEECallOrigin::ensure_origin(origin)?;
+			let mut id_graph_links =
+				IDGraphs::<T>::get(&who).ok_or(Error::<T>::IdentityNotFound)?;
+			let id_graph_link = id_graph_links
+				.iter_mut()
+				.find(|(id, _)| id == &identity)
+				.ok_or(Error::<T>::IdentityNotFound)?;
+			id_graph_link.1.set_web3networks(web3networks.clone());
+
+			IDGraphs::<T>::insert(who.clone(), id_graph_links);
+
+			let identity_hash = H256::from(blake2_256(&identity.encode()));
+
+			Self::deposit_event(Event::Web3NetworksUpdated {
+				identity: identity_hash,
+				web3networks,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn remove_identity(
+			origin: OriginFor<T>,
+			who: Identity,
+			identities: Vec<OmniAccountIdentity>,
+		) -> DispatchResult {
+			let _ = T::TEECallOrigin::ensure_origin(origin)?;
+
+			let mut id_graph_links =
+				IDGraphs::<T>::get(&who).ok_or(Error::<T>::IdentityNotFound)?;
+
+			if !identities.is_empty() {
+				id_graph_links.retain(|(id, _)| {
+					if identities.contains(id) {
+						let identity_hash = H256::from(blake2_256(&id.encode()));
+						LinkedIdentities::<T>::remove(identity_hash);
+						false
+					} else {
+						true
+					}
+				});
+				IDGraphs::<T>::insert(who.clone(), id_graph_links);
+			} else {
+				id_graph_links.iter().for_each(|(id, _)| {
+					let identity_hash = H256::from(blake2_256(&id.encode()));
+					LinkedIdentities::<T>::remove(identity_hash);
+				});
+				IDGraphs::<T>::remove(&who);
+			}
+
+			Self::deposit_event(Event::IdentityRemoved { who, identities });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(5)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn make_identity_public(
+			origin: OriginFor<T>,
+			who: Identity,
+			private_identity: OmniAccountIdentity,
+			public_identity: OmniAccountIdentity,
```

Comment:
It seems we only need the `public_identity` here?  `private_identity` can be derived from public identity?

Reply:
Yes I'll remove this together with the changes to send the identity_hash from the worker